harness/determined/launch/deepspeed.py

```diff
@@ -86,7 +70,7 @@ def create_hostlist_file(
     if len(ip_addresses) == 1:
         trial_runner_hosts[0] = "localhost"
 
-    if hostfile_path is not None:
+    if hostfile_path is not None and not os.path.exists(hostfile_path):
```

Comment:
should we just throw an exception here if the hostfile path already exists? if it's unique to allocation id, this should never happen right?

Reply:
it's for the multi-node shared fs case where if the hostfile has already been created by one node, then other nodes don't need to create it anymore.